### PR TITLE
Feat b y intensity ratio

### PR DESCRIPTION
**Pull request by JemmaLDaniel**
### Summary

Adds a new `b_y_intensity_ratio` feature to `FragmentMatchFeatures` (and its chimeric counterpart `chimeric_b_y_intensity_ratio` to `ChimericFeatures`). This feature captures the ratio of total matched b-ion intensity to total matched y-ion intensity in the observed spectrum, providing the calibrator with information about fragmentation directionality — a useful discriminator since correctly identified peptides tend to exhibit characteristic b/y intensity patterns depending on fragmentation type, enzyme, charge state, and sequence properties.

### Changes

- **`winnow/calibration/features/utils.py`**:
  - `find_matching_ions` now returns per-ion matched intensities (including isotopic envelope contributions) as a fifth return value.
  - `compute_ion_identifications` aggregates matched ion intensities by ion type (b vs y) and computes their ratio, returning `b_y_intensity_ratio` as a new output.
  - Adds `compute_b_y_intensity_ratio` helper that sums intensities for b-ions and y-ions from the matched annotations and computes their ratio (returns 0.0 when no y-ion intensity is observed).
- **`winnow/calibration/features/fragment_match.py`** — adds `b_y_intensity_ratio` to `columns()` and stores the computed value in `dataset.metadata`.
- **`winnow/calibration/features/chimeric.py`** — adds `chimeric_b_y_intensity_ratio` column; removes unused commented-out code for an earlier spectrum match quality approach.
- **`docs/api/features/fragment_match.md`**, **`docs/api/features/chimeric.md`** — document the new feature.
- **`tests/calibration/features/test_utils.py`** — tests for `compute_b_y_intensity_ratio` covering standard cases, edge cases (no y-ions, no b-ions, empty input), and integration with `find_matching_ions`.
- **`tests/calibration/features/test_fragment_match.py`**, **`test_chimeric.py`** — assert new columns are present in computed outputs.

### CODE REVIEW

File: tests/calibration/features/test_utils.py [hunk 1]
```diff
@@ -55,7 +56,7 @@ def test_find_matching_ions_with_tolerance(self):
         target_intensities = [1000.0, 2000.0]
         tolerance = 0.02
 
-        match_fraction, average_intensity, _, _ = find_matching_ions(
+        match_fraction, average_intensity, *_ = find_matching_ions(
```
Comment (BioGeek) [inline tests/calibration/features/test_utils.py]: Cool! I didn't knew you could do `*_`!

File: winnow/calibration/features/utils.py [hunk 2]
```diff
@@ -320,13 +325,18 @@ def find_matching_ions(
...
+            # Record total intensity for this ion (M0 + isotopes)
+            matched_ion_intensities.append(ion_intensity)
 
     return (
         num_matches / len(source_mz),
-        match_intensity / total_target_intensity,
+        matched_target_intensity / total_target_intensity,
         matched_ion_annotations,
         matched_ion_mz,
+        matched_ion_intensities,
```
Comment (BioGeek) [inline winnow/calibration/features/utils.py:477]: When existing downstream code imports this backwards-compatible helper and unpacks the documented four values, adding this fifth tuple element makes calls like `rate`, `intensity`,  `annotations `, `mz = find_matching_ions(...)` fail with `ValueError: too many values to unpack`. Raise warning instead that old format is no longer supported.

File: winnow/calibration/features/utils.py [hunk 3]
```diff
@@ -388,6 +404,7 @@ def compute_ion_identifications(
                 longest_y_series,
                 complementary_ion_count,
                 max_ion_gap,
+                b_y_intensity_ratio,
```
Comment (BioGeek) [inline winnow/calibration/features/utils.py:553]: Same `ValueError: too many values to unpack` remark here. raise a warning that old format is no longer supported.

**Review by BioGeek — CHANGES_REQUESTED** [review]
Again mostly backwards compatability remarks.

File: winnow/calibration/features/utils.py (hunk 2 above)
Comment (JemmaLDaniel) [reply winnow/calibration/features/utils.py:477]: I've switched to returning a named tuple `IonMatchResult` from `find_matching_ions` so that callers can use `result.match_rate` etc, and I can add additional fields in the later PRs without breaking attribute access. You can also use the star-unpack (`*_`) to handle positional unpacking over multiple trailing arguments. Then, I've also added a `DeprecationWarning` suggesting users to migrate. Shout if there are more issues with this version.

**Review by BioGeek — APPROVED** [review]
I've made some [changes](https://github.com/instadeepai/winnow/pull/184/changes/c89819e5a8b05210a317e0442851185e10e70baa) so that positional unpacking now fails deliberately with an upgrade message:

  - `winnow/calibration/features/utils.py:25`: `IonMatchResult.__iter__` raises `TypeError` telling callers to use named fields.
  - `winnow/calibration/features/utils.py:44`: added `IonIdentificationResult` for `compute_ion_identifications`, also raising on positional unpacking.
  - Updated `FragmentMatchFeatures` and `ChimericFeatures` to consume named fields instead of unpacking.
  - Exported `IonIdentificationResult` from both feature API modules.
  - Updated tests to assert the new explicit error instead of backwards-compatible unpacking.

Is good to merge now!
